floodlight/io/dfl.py

```diff
@@ -342,24 +281,120 @@ def read_pitch_from_mat_info_xml(filepath_mat_info: Union[str, Path]) -> Pitch:
+        elif team_info.get("TeamId") == away_id:
+            team = "Away"
+        else:
+            team = None
+
+        # skip referees sometimes referred to as a team in new data formats
+        if team not in ["Home", "Away"]:
+            continue
 
-    This function provides a high-level access to the particular DFL Match Event feed
-    and returns Event objects for both teams. The number of segments is inferred from
-    the data, yet data for each segment is stored in a separate object.
+        # create teamsheets
+        teamsheets[team]["player"] = [
+            player.get("Shortname") for player in team_info.find("Players")
+        ]
+        teamsheets[team]["pID"] = [
+            player.get("PersonId") for player in team_info.find("Players")
+        ]
+        teamsheets[team]["jID"] = [
+            int(player.get("ShirtNumber")) for player in team_info.find("Players")
+        ]
+        teamsheets[team]["position"] = [
+            player.get("PlayingPosition") for player in team_info.find("Players")
+        ]
+        teamsheets[team]["tID"] = team_info.get("TeamId")
+        teamsheets[team]["team"] = team_info.get("TeamName")
+
+    # create teamsheet objects
+    for team in teamsheets:
+        teamsheets[team] = Teamsheet(teamsheets[team])
+
+    return teamsheets
+
+
+def read_event_data_xml(
+    filepath_events: Union[str, Path],
+    filepath_mat_info: Union[str, Path],
+    home_teamsheet: Teamsheet = None,
+    away_teamsheet: Teamsheet = None,
+) -> Tuple[Events, Events, Events, Events, Teamsheet, Teamsheet]:
+    """Parses a DFL Match Event XML file and extracts the event data as well as
+    teamsheets.
+
+    The structure of the official tracking system of the DFL (German Football League)
+    contains two separate xml files, one containing the actual data as well as a
+    metadata file containing information about teams, pitch size, and start- and
+    endframes of match periods. This function provides high-level access to DFL data by
+    parsing "the full match" and returning Events-objects parsed from the event data
+    xml-file as well as Teamsheet-objects parsed from the metadata xml-file. The number
+    of segments is inferred from the data, yet data for each segment is stored in a
+    separate object.
 
     Parameters
     ----------
     filepath_events: str or pathlib.Path
         Full path to XML File where the Event data in DFL format is saved.
+    filepath_mat_info: str or pathlib.Path
+        Full path to XML File where the Match Information data in DFL format is saved.
+    home_teamsheet: Teamsheet, optional
+        Teamsheet-object for the home team used to create link dictionaries of the form
+        `links[team][jID] = xID` and  `links[team][pID] = jID`. The links are used to
+        map players to a specific xID in the respective XY objects. Should be supplied
+        if that order matters. If given as None (default), teamsheet is extracted from
```

Comment:
```suggestion
        for custom xID ordering. If given as None (default), teamsheet is extracted from
```

Reply:
Maybe one additional sentence on the automatic creation of xIDs would be helpful, either here or in the Notes. Meaning: if no info is provided, how are xIDs assigned? Probably close to randomness, but maybe we can state something like "in order of appearance" or "with ascending jIDs"

Reply:
This comment also applies to the xy data parser. I also noticed you never invoke the ``add_xIDs()`` method in this parser (you don't exactly need it) so the explanation of xIDs is actually not correct as the teamsheet you return probably doesn't even have an xID column (if I dont miss anything). Any thoughts on this?
a) Include xIDs anyhow for consistency, but risk overwriting/or causing confusion
b) Don't include xIDs and change docstring, but both parsers return "different" teamsheet objects

Reply:
Decided for b) to be the less confusing option.

Reply:
Yep, agree